### ISSUE TITLE
mock_s3 breaks delete_objects with Unicode

### ISSUE THREAD

**Issue by samuelfekete** [spulec/moto#1198]
## Importing mock_s3 causes Unicode errors with delete_objects

Steps to reproduce (Python 2.7):
```
pip install boto3    # (1.4.7)
pip install moto     # (1.1.14)
python
>>> import boto3
>>> from moto import mock_s3
>>> client = boto3.client('s3')
>>> client.delete_objects(Bucket='bucket-name', Delete={'Objects': [{'Key': u'ó'}]})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../local/lib/python2.7/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File ".../local/lib/python2.7/site-packages/botocore/client.py", line 588, in _make_api_call
    operation_model, request_dict)
  File ".../local/lib/python2.7/site-packages/botocore/endpoint.py", line 141, in make_request
    return self._send_request(request_dict, operation_model)
  File ".../local/lib/python2.7/site-packages/botocore/endpoint.py", line 170, in _send_request
    success_response, exception):
  File ".../local/lib/python2.7/site-packages/botocore/endpoint.py", line 249, in _needs_retry
    caught_exception=caught_exception, request_dict=request_dict)
  File ".../local/lib/python2.7/site-packages/botocore/hooks.py", line 227, in emit
    return self._emit(event_name, kwargs)
  File ".../local/lib/python2.7/site-packages/botocore/hooks.py", line 210, in _emit
    response = handler(**kwargs)
  File ".../local/lib/python2.7/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File ".../local/lib/python2.7/site-packages/botocore/retryhandler.py", line 251, in __call__
    caught_exception)
  File ".../local/lib/python2.7/site-packages/botocore/retryhandler.py", line 269, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File ".../local/lib/python2.7/site-packages/botocore/retryhandler.py", line 317, in __call__
    caught_exception)
  File ".../local/lib/python2.7/site-packages/botocore/retryhandler.py", line 223, in __call__
    attempt_number, caught_exception)
  File ".../local/lib/python2.7/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 69: ordinal not in range(128)
```

https://github.com/boto/botocore/issues/1281

**Comment by JackDanger** [comment]
Thanks for reporting this @samuelfekete! I've fixed it in #1199 and will merge when green.

**Comment by samuelfekete** [comment]
Thanks @JackDanger for fixing it so quickly!
